### PR TITLE
Add two sniffs to find and remove unused "use" clauses

### PR DESCRIPTION
**Pull request by thiemowmde**
A run on the Wikibase.git code base currently shows 9 violations that are all fine and can be fixed automatically.

Note that the Wikibase.Namespaces.UnusedUse sniff does have a series of known as well as potential flaws:
* It considers class names aliased with `as`. The code to find the alias might be a bit naive and miss edge cases. However, I can not think of a test case that fails. Can you find one?
* It only considers a list of PHPDoc tags that are known to contain class names: `@expectedException`, `@param`, `@return`, `@throw`, and `@var` (as well as `@type` for compatibility, even if deprecated, see #8). Did I missed one?

### CODE REVIEW

**Review by lucaswerkmeister — APPROVED** [review]
I tried this out in the WikibaseQualityConstraints extension, where it removes 92 lines (all correct, as far as I can tell – I’ve just been sloppy with the imports). Seems to work fine.

**Review by lucaswerkmeister — CHANGES_REQUESTED** [review]
Found a bug:
```php
<?php
use A\B as /*_*/C;
$b = new C();
```
turns into
```php
<?php
C;
$b = new C();
```
which is invalid.

**Comment by addshore** [comment]
Can this make its way into the mediawiki codesniffer?

**Comment by thiemowmde** [comment]
@lucaswerkmeister: Good catch. That was easy to fix.

@addshore: See https://gerrit.wikimedia.org/r/375547. Both teams started to work on the same idea independently. Naturally, I like my version more. ;-)

**Comment by lucaswerkmeister** [comment]
Sorry, there’s another broken case:
```php
<?php
use A\B as /** @var */ C;

$b = new C();
```
I tried expanding your fix, but it doesn’t work:
```patch
diff --git a/Wikibase/Sniffs/Namespaces/UnusedUseSniff.php b/Wikibase/Sniffs/Namespaces/UnusedUseSniff.php
index f2e94c4737..09dc891d09 100644
--- a/Wikibase/Sniffs/Namespaces/UnusedUseSniff.php
+++ b/Wikibase/Sniffs/Namespaces/UnusedUseSniff.php
@@ -24,6 +24,10 @@ public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
 		$useEndPtr = $phpcsFile->findNext( [
 			T_AS,
 			T_COMMENT,
+			T_DOC_COMMENT_OPEN_TAG,
+			T_DOC_COMMENT_CLOSE_TAG,
+			T_DOC_COMMENT_STRING,
+			T_DOC_COMMENT_TAG,
 			T_NS_SEPARATOR,
 			T_STRING,
 			T_WHITESPACE,
```
But this is a seriously pathological case… if the bug is too hard to fix, I think we could merge this without the bugfix.

**Comment by thiemowmde** [comment]
Solved it. Thanks for poking this stuff!

**Comment by lucaswerkmeister** [comment]
Great, thanks! That fix also looks more robust ;)

**Review by lucaswerkmeister — APPROVED** [review]
